### PR TITLE
Use official maven docker image

### PR DESCRIPTION
**Pull request by hgeraldino**
To help us review your pull request, please consider providing an overview of the following:
* What is the type of the change (bug fix, feature, documentation and etc.) ?
  Updates the base image used to build the java docker container. Also, installs `protoc` from the official repos.
    
* What are the current behavior and expected behavior, if this is a bugfix ?
* What are the steps required to reproduce the bug, if this is a bugfix ?
* What is the current behavior and new behavior, if this is a feature change or enhancement ?
* [Optional] Why is the new behavior better than the current behavior, if this is a feature change ?
  I'd argue it is, as it uses more recent versions of both `maven` and `protoc`.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0145711c/smoketest.log'>Error &#x26A0; </a>.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0145711c/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0145711c/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2637'>Resubmit</a></p>

File: contrib/docker/Dockerfile.jdbc.build [hunk 1]
```diff
@@ -1,18 +1,10 @@
...
-  mv /bin/apache-maven-3.5.0 bin/maven
+RUN apt-get update \
+    && apt-get install -y protobuf-compiler  \
+    && apt-get clean
 
 # Sets maven to use a useful settings.xml file
 COPY \
   contrib/docker/maven-settings.xml /bin/maven/conf/settings.xml
 
-# Install protobuf 3.2 and give all users exec access
```
Comment (riverszhang89) [inline contrib/docker/Dockerfile.jdbc.build:13]: I think the version of the protobuf compiler needs to match [the version in the pom file](https://github.com/bloomberg/comdb2/blob/master/cdb2jdbc/pom.xml#L30). Protobuf isn't particularly good at backward compatibility. See the build failure [here](http://comdb2.s3-website-us-east-1.amazonaws.com/tests/0145711c/smoketest.log).

Comment (hgeraldino) [reply contrib/docker/Dockerfile.jdbc.build:13]: Noted.

I bumped the library version to 3.8.0.

Comment (riverszhang89) [reply contrib/docker/Dockerfile.jdbc.build:13]: I actually prefer to lock the version of protobuf. The version of protobuf-compiler from apt-get may very well change in the future. It'll then likely break the build again. With that said, I'm fine with using maven from apt-get (it didn't exist when the Dockerfile was first written), and leaving the protobuf installation as is.

**Pull request closed without merging**